### PR TITLE
Rename OPENSSL_CTX prefix to OSSL_CTX

### PR DESCRIPTION
**Pull request by mspncp**
For an explanation, see pull request #12621.

### CODE REVIEW

**Comment by mspncp** [comment]
I force-pushed a fix, because I forgot about the corresponding renamings of prefixes in small letters. Here is the updated script.

```bash
#!/bin/bash

set -o errexit

# this script looks more complicated as it is, because it does some
# parameter checking and it can be rerun several times.
#
# The actual renaming consists of only the last four lines of code.
#
# Every time the script is run it will redo the changes at the current
# tip of the master branch.

FROM=$1
TO=$2

function usage () {
	cat  >& 2 <<EOF
usage: one of the following two commands
    ./rename-prefix OPENSSL_CTX OSSL_CTX
    ./rename-prefix OPENSSL_CTX OSSL_CONTEXT
EOF
	exit 1
}

case $FROM in
	OPENSSL_CTX)
		;;
	*)
		usage
		;;
esac

# add necessary padding manually (I had no ambitions to figure out the magic bash string calculations)
case $TO in
	OSSL_CTX)
		from_pad=''
		to_pad='   '
	;;
	OSSL_CONTEXT)
		from_pad=' '
		to_pad=''
		;;
	*)
		usage
esac

from=${FROM,,}
to=${TO,,}

branch=pr-rename-$FROM-to-$TO

# reuse existing branch or create a new one if it does not exist yet
if git show-ref --verify --quiet refs/heads/$branch; then
	git checkout $branch
	git reset --hard master
else
	git checkout -b $branch master
fi


#
# the renaming starts here
#


# treat util/libcrypto.num and util/other.syms specially, to get the alignment correct
git grep -l ${FROM} -- util/ | xargs sed -i -E "s/${FROM}([a-zA-Z0-9_]*)${from_pad}/${TO}\1${to_pad}/g"

# treat util/libcrypto.num and util/other.syms specially, to get the alignment right
git grep -il ${FROM} | xargs sed -i "s/${FROM}/${TO}/g;s/${from}/${to}/g"

# rename the only file which contains '${FROM}' in its file name
git mv doc/man3/${FROM}.pod doc/man3/OSSL_CTX.pod

# commit the changes
git commit -a -m "Rename ${FROM} prefix to ${TO}"

```

File: crypto/evp/evp_fetch.c [hunk 1]
```diff
@@ -67,9 +67,9 @@ static void *alloc_tmp_evp_method_store(OPENSSL_CTX *ctx)
         ossl_method_store_free(store);
 }
 
-static OSSL_METHOD_STORE *get_evp_method_store(OPENSSL_CTX *libctx)
+static OSSL_METHOD_STORE *get_evp_method_store(OSSL_CTX *libctx)
 {
-    return openssl_ctx_get_data(libctx, OPENSSL_CTX_EVP_METHOD_STORE_INDEX,
+    return ossl_ctx_get_data(libctx, OSSL_CTX_EVP_METHOD_STORE_INDEX,
                                 &evp_method_store_method);
```
Comment (paulidale) [inline crypto/evp/evp_fetch.c:73]: Nit: indentation.

File: crypto/provider_core.c [hunk 2]
```diff
@@ -179,23 +179,23 @@ static void *provider_store_new(OPENSSL_CTX *ctx)
...
 };
 
-static struct provider_store_st *get_provider_store(OPENSSL_CTX *libctx)
+static struct provider_store_st *get_provider_store(OSSL_CTX *libctx)
 {
     struct provider_store_st *store = NULL;
 
-    store = openssl_ctx_get_data(libctx, OPENSSL_CTX_PROVIDER_STORE_INDEX,
+    store = ossl_ctx_get_data(libctx, OSSL_CTX_PROVIDER_STORE_INDEX,
                                  &provider_store_method);
```
Comment (paulidale) [inline crypto/provider_core.c:192]: Indentation.

File: doc/internal/man3/openssl_ctx_get_data.pod [hunk 3]
```diff
@@ -2,29 +2,29 @@
...
+ typedef struct ossl_ctx_method {
+     void *(*new_func)(OSSL_CTX *ctx);
      void (*free_func)(void *);
- } OPENSSL_CTX_METHOD;
+ } OSSL_CTX_METHOD;
 
- void *openssl_ctx_get_data(OPENSSL_CTX *ctx, int index,
-                            const OPENSSL_CTX_METHOD *meth);
+ void *ossl_ctx_get_data(OSSL_CTX *ctx, int index,
+                            const OSSL_CTX_METHOD *meth);
```
Comment (paulidale) [inline doc/internal/man3/openssl_ctx_get_data.pod:19]: indentation.

File: doc/internal/man3/openssl_ctx_get_data.pod [hunk 4]
```diff
@@ -2,29 +2,29 @@
...
 
- int openssl_ctx_run_once(OPENSSL_CTX *ctx, unsigned int idx,
-                          openssl_ctx_run_once_fn run_once_fn);
- int openssl_ctx_onfree(OPENSSL_CTX *ctx, openssl_ctx_onfree_fn onfreefn);
+ int ossl_ctx_run_once(OSSL_CTX *ctx, unsigned int idx,
+                          ossl_ctx_run_once_fn run_once_fn);
```
Comment (paulidale) [inline doc/internal/man3/openssl_ctx_get_data.pod:22]: indentation

File: doc/internal/man3/openssl_ctx_get_data.pod [hunk 5]
```diff
@@ -99,26 +99,26 @@ To get and use the data stored in the library context, simply do this:
...
+     return ossl_ctx_onfree(ctx, &foo_cleanup)
  }
 
- int foo_some_function(OPENSSL_CTX *ctx)
+ int foo_some_function(OSSL_CTX *ctx)
  {
-    if (!openssl_ctx_run_once(ctx,
-                              OPENSSL_CTX_FOO_RUN_ONCE_INDEX,
+    if (!ossl_ctx_run_once(ctx,
+                              OSSL_CTX_FOO_RUN_ONCE_INDEX,
```
Comment (paulidale) [inline doc/internal/man3/openssl_ctx_get_data.pod:121]: indentation

File: include/internal/cryptlib.h [hunk 6]
```diff
@@ -138,61 +138,61 @@ typedef struct ossl_ex_data_global_st {
...
-int openssl_ctx_is_global_default(OPENSSL_CTX *ctx);
+OSSL_CTX *ossl_ctx_get_concrete(OSSL_CTX *ctx);
+int ossl_ctx_is_default(OSSL_CTX *ctx);
+int ossl_ctx_is_global_default(OSSL_CTX *ctx);
 
 /* Functions to retrieve pointers to data by index */
-void *openssl_ctx_get_data(OPENSSL_CTX *, int /* index */,
-                           const OPENSSL_CTX_METHOD * ctx);
+void *ossl_ctx_get_data(OSSL_CTX *, int /* index */,
+                           const OSSL_CTX_METHOD * ctx);
```
Comment (paulidale) [inline include/internal/cryptlib.h:176]: indentation

**Comment by mspncp** [comment]
> Indentation issues, I doubt I flagged them all.

Hmmm... those issues appeared only after I added the renaming of the lower case prefixes. Never mind, if there are not too many of them, I could collect them in a fixup patch which can be applied automatically and (hopefully) without conflicts.

> Rewrapping shorter lines isn't a style break.

Does this comment, together with your choice of reviewing #12622 (instead of #12621) already indicate a preference?

**Comment by mspncp** [comment]
Closing in favour of #12621 (which happens to be a palindrome).

**Pull request closed without merging**